### DISCUSSION TITLE
Python override Windows HOME for user data

### DISCUSSION THREAD

**Discussion by Alexpux** [msys2/MINGW-packages#7646] [Ideas]
For the continue of https://github.com/msys2/MINGW-packages/discussions/7619 I have idea to override Windows HOME folder for user data.
Possible values are:

- /mingw{32,64}/home/$USER>
- /mingw64/home-python-pip-$USER

I want to hear what Python users think about it? It can solve problem with co-existing with official Python installations and do Mingw python some more possibilies to relocate when move folder to other PC.

**Comment by lazka** [comment]
I assume this is for `pip install --user` aka the user install scheme? Why not use the unix variant "~/.local/..." and encode MSYSTEM there?

**Reply by naveen521kk** [reply]
I think it uses Linux variant there.
```sh
$ python -m site --user-base
C:\Users\User/.local
```
While MSYSTEM is something which isn't enoded though.

**Comment by mingwandroid** [comment]
We are concerned about users wishing to coinstall various pythons and not having them conflict.. now given we adopt fhs layout for ours that may not be such a problem but we might also want to consider avoiding conflicts with e.g. cygwin python or msys2 python.

**Reply by mingwandroid** [reply]
We would hard-code the Python implementation's idea of what MSYSTEM should be too I guess, for people avoiding running via bash.

**Comment by Alexpux** [comment]
My idea also that Python will work with its own home even if it using not from MSYS2 shell

**Comment by naveen521kk** [comment]
This idea looks like https://github.com/msys2/MINGW-packages/issues/7564

**Reply by mingwandroid** [reply]
.. kinda, but not about TEMP (which is about the arbitrary stuff) but about site-packages (which touches on ABI compatibility)

**Comment by mingwandroid** [comment]
I'm liking the discussions thing btw! It's really good to get an official discussion and record of our thoughts and plans before we commence work!

**Comment by naveen521kk** [comment]
Also, the below code which currently returns, Windows Home, should return msys2 home.
```py
>>> os.path.expanduser('~')
'C:\\Users\\User'
```

**Reply by Alexpux** [reply]
Well I think we MUST not use MSYS2 home for MINGW packages. For mingw programs ROOT is ${MINGW_PREFIX}

**Reply by mingwandroid** [reply]
Let's try to keep HOME as a separate concern from where MSYS2/mingw-w64 python stores its local site-packages. I am somewhat inclined to agree with @lazka's idea of keeping it in` $HOME/.local/<HARDCODED_IN_PYTHON_EXE_SUBSYSTEM_NAME>` (then what we do about `$HOME`, we decide separately).

**Reply by naveen521kk** [reply]
> Well I think we MUST not use MSYS2 home for MINGW packages. For mingw programs ROOT is ${MINGW_PREFIX}

I am sorry, I am quite new to this. But I meant exactly the same as you told, ie: `C:\msys64\home\User`
No wait am I confused.

**Reply by mingwandroid** [reply]
I know I said I'd keep it separate, but thinking out loud, I think I'd like mingw32 and mingw64 home to be controllable same as MSYS2's home in /etc/nsswitch.conf

**Reply by mingwandroid** [reply]
@naveen521kk check /etc/nsswitch, it's variable. I set my HOMEs all to the Windows dir FWIW, always have done.
